File: prime_number_check.java
import java.util.Scanner; 
public class PrimeCheck
{
    public static void main(String args[])
    {
        //declaration 
        Scanner sc = new Scanner(System.in); 
        int n ; 
        boolean flag = false; 
        
        // prompting and accpeting the value from the user
        System.out.print("Enter number: ");
        n = sc.nextInt(); 
        
        // finding for factors 
        for( int i = 2 ; i < n ; i++)
        {
            if( n % i == 0)
            {
                flag = true;
                break;
            }
        }
        
        if(flag)
        {
            System.out.println(n + " is not a prime number");
        }
        else
        {
            System.out.println(n + " is a prime number");
        }
    }
}

File: prime_number_check.py
# initialization
flag = False

# accepting the number from the user
number = input("Enter a number: ")

# loop to find factors
for i in range(2, (int(number)-1)):
    if int(number) % i == 0:
        flag = True
        break

# checking if there are any factors
if flag:
    print(number + " is not a prime number")
else:
    print(number + " is a prime number")
